### DISCUSSION TITLE
Fooocus New update (Add Object new option VS Inpaint)

### DISCUSSION THREAD

**Discussion by EKKIV0K** [lllyasviel/Fooocus#989] [Q&A]
hello guys i'm writing this thread cause i am a little bit disapointed, 
But i will do as quick as possible, Before we had inpain and outpaint and on this option, we had the possibility to add object.
my question is, what's new, what is the difference bet ween inpaint & outpaint and the option "add object, background etc....) ?!
cause it was already possible with the old inpaint outpaint option :/

**Comment by GodwinF** [comment]
Not sure what those options do, we need some examples please.

**Comment by lllyasviel** [comment]
hi please refer to the new document.

the default inpaint is in line with midjourney's inpaint, which is basically a "regional vary" tool. It is good at making variations of local objects but not very good at adding large objects.

but some users want to add large objects and in those cases the newer mode "adding object" could be more suitable

note that we do not invent this: the multi-mode things is from Adobe Generative Fill

**Comment by EKKIV0K** [comment]
@lllyasviel good thing ! really but i noticed that the regular method even for large object is better cause it look like to foolow more the color / lighting / shadow of the original image. The new method seems to lighting the inserted object much more, and it result of something not fitting with original image :/
